src/sinks/datadog/logs/tests.rs

```diff
@@ -0,0 +1,233 @@
+use crate::sinks::datadog::logs::DatadogLogsConfig;
+use crate::{
+    config::SinkConfig,
+    sinks::util::test::{build_test_server_status, load_sink},
+    test_util::{next_addr, random_lines_with_stream},
+};
+use bytes::Bytes;
+use futures::SinkExt;
+use futures::{
+    channel::mpsc::{Receiver, TryRecvError},
+    stream, StreamExt,
+};
+use hyper::StatusCode;
+use indoc::indoc;
+use pretty_assertions::assert_eq;
+use std::sync::Arc;
+use vector_core::event::Event;
+use vector_core::event::{BatchNotifier, BatchStatus};
+
+#[test]
+fn generate_config() {
+    crate::test_util::test_generate_config::<DatadogLogsConfig>();
+}
```

Comment:
I could see this test living in `config.rs` and `test.rs` just having the integration tests.

Reply:
Ah yeah, I did just kind of move all the tests together in one block. I'll scoot them around to their appropriate homes.